File: 2021/python/day1.py
from statistics import mean

# Read file as list
with open("../data/input_1.txt") as file:
   depths = file.read().splitlines()
depths = list(map(int, depths))

# PART 1
# Create tuple pairs and iterate
depth_tuples = list(zip(depths[:-1], depths[1:]))
depth_inc_counter = 0    
for i in depth_tuples:
    if i[1] > i[0]:
        depth_inc_counter += 1

print("Answer to part 1 is: {}".format(depth_inc_counter))

# Create three-measurement window pairs in tuples
windows = list(zip(depths[:-2], depths[1:-1], depths[2:]))
depth_window_tuples = list(zip(windows[:-1], windows[1:]))

depth_inc_counter = 0    
for i in depth_window_tuples:
    if mean(i[1]) > mean(i[0]):
        depth_inc_counter += 1

print("Answer to part 2 is: {}".format(depth_inc_counter))

File: 2021/julia/day1.jl
import Statistics

# Read in file
depths = zeros(Int64, 0)
open("../data/input_1.txt") do file
    s = read(file, String)
    append!(depths, parse.(Int, split(s, r"\n|\r\n")))
end

# PART 1
depth_inc_counter = 0
for i in zip(depths[1:end-1],depths[2:end])
    if i[2] > i[1]
        global depth_inc_counter += 1
    end
end
println("Answer to part 1 is: ", depth_inc_counter)

# PART 2
windows = collect(zip(depths[1:end-2], depths[2:end-1], depths[3:end]))
depth_window_tuples = zip(windows[1:end-1], windows[2:end])

depth_inc_counter = 0    
for i in depth_window_tuples
    if Statistics.mean(i[2]) > Statistics.mean(i[1])
        global depth_inc_counter += 1
    end
end

println("Answer to part 2 is: ", depth_inc_counter)